sarif/sarif_file_utils.py

```diff
@@ -100,6 +100,103 @@ def read_result_location(result) -> Tuple[str, str]:
     return (file_path, line_number)
 
 
+def read_result_rule(result, run) -> Tuple[Union[dict, None], int]:
+    """
+    Return's the corresponding rule object for the specified result, plus its index
+    in the rules array. Follows the rules at
+    https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html#_Toc141790895
+    """
+    ruleIndex = result.get("ruleIndex", None)
+    ruleId = result.get("ruleId", None)
+    rule = result.get("rule", None)
+
+    if not ruleIndex and rule:
```

Comment:
Careful here, `ruleIndex` 0 and None will both make `not ruleIndex` evaluate to `True`, but I think you need to distinguish them.  Suggestion that would cover L113-117:

```python
if rule:
  if ruleIndex is None:
    ruleIndex = rule.get("index")
  if ruleId is None:
    ruleId = rule.get("id")
```

Reply:
Thank you. For some reason I was assuming that these values would always be strings (serialized ints).